dags/hivemind_etl_helpers/src/db/github/extract/issues.py

```diff
@@ -2,96 +2,136 @@
+            i.repository_id IN $repoIds
+        """
+        if from_date is not None:
+            query += "AND datetime(i.updated_at) >= datetime($from_date)"
+
+        query += """
+            MATCH (repo:Repository {id: i.repository_id})
+            RETURN
+                user.login as author_name,
+                i.id as id,
+                i.title as title,
+                i.body as text,
+                i.state as state,
+                i.state_reason as state_reason,
+                i.created_at as created_at,
+                i.updated_at as updated_at,
+                i.closed_at as closed_at,
+                i.latestSavedAt as latest_saved_at,
+                i.html_url as url,
+                i.repository_id as repository_id,
+                repo.full_name as repository_name
+            ORDER BY datetime(created_at)
+        """
+
+        def _exec_query(tx, repoIds, from_date):
+            result = tx.run(query, repoIds=repoIds, from_date=from_date)
+            return list(result)
+
+        with self.neo4j_driver.session() as session:
+            raw_records = session.execute_read(
+                _exec_query,
+                repoIds=repository_id,
+                from_date=from_date,
+            )
+
+        return raw_records
+
+    def fetch_issues(
+        self,
+        repository_id: list[int],
+        from_date: datetime | None = None,
+    ) -> list[GitHubIssue]:
+        """
+        fetch issues from data dump in neo4j
+
+        Parameters
+        ------------
+        repository_id : list[int]
+            a list of repository id to fetch their issues
+        from_date : datetime | None
+            get the issues form a specific date that they were created
+            default is `None`, meaning to apply no filtering on data
+
+        Returns
+        --------
+        github_issues : list[GitHubIssue]
+            list of neo4j records as the extracted issues
+        """
+        records = self._fetch_raw_issues(repository_id, from_date)
+
+        github_issues: list[GitHubIssue] = []
+        for record in records:
+            issue = GitHubIssue.from_dict(record)
+            github_issues.append(issue)
+
+        return github_issues
+
+    def fetch_issue_ids(
+        self,
+        repository_id: list[int],
+        from_date: datetime | None = None,
+    ) -> list[GitHubIssueID]:
+        """
+        fetch issues from data dump in neo4j
+
+        Parameters
+        ------------
+        repository_id : list[int]
+            a list of repository id to fetch their issues
+        from_date : datetime | None
+            get the issues form a specific date that they were created
+            default is `None`, meaning to apply no filtering on data
+
+        Returns
+        --------
+        github_issues_ids : list[GitHubIssueID]
+            list of neo4j records as the extracted issue ids
+        """
+        records = self._fetch_raw_issues(repository_id, from_date)
```

Comment:
Could you write another function for just fetching the ids? Fetching the whole data and just using their ids is inefficient. You can name the method as `_fetch_raw_issue_ids` Please let me know if you need any help on the neo4j query.